src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -44,6 +44,26 @@ public IOperation Create(BoundNode boundNode)
 
         private IOperation CreateInternal(BoundNode boundNode)
         {
+            IOperation operation = CreateInternalCore(boundNode);
+
+            // If the syntax for the bound node is parenthesized, walk up the parentheses wrapping the operation with ParenthesizedExpression operation nodes.
+            // Compiler generated and lowered bound nodes shouldn't be parenthesized.
+            if (!boundNode.WasCompilerGenerated &&
+                boundNode.Syntax.Parent is ParenthesizedExpressionSyntax parenthesizedSyntax)
+            {
+                do
+                {
+                    operation = new ParenthesizedExpression(operation, _semanticModel, parenthesizedSyntax, operation.Type, operation.ConstantValue, isImplicit: false);
+                    parenthesizedSyntax = parenthesizedSyntax.Parent as ParenthesizedExpressionSyntax;
+                }
+                while (parenthesizedSyntax != null);
+            }
```

Comment:
Consider writing loop with one check for ParenthesizedExpressionSyntax rather than two (in `if` and `while`) and compare `Syntax.Kind` rather than `as`.

Reply:
Sure